contracts/mosaic/MerklePatriciaProof.sol

```diff
@@ -18,127 +14,79 @@ library MerklePatriciaProof {
      * @param root The root hash of the trie.
      * @return The boolean validity of the proof.
      */
-    function verify(bytes32 value, bytes encodedPath, bytes rlpParentNodes, bytes32 root) internal pure returns (bool) {
+    function verify(bytes32 value, bytes addr, bytes rlpParentNodes, bytes32 root) public returns (bool) {
+
+        //Convert back RLP encoded parent Nodes to list of Nodes
         RLP.RLPItem memory item = RLP.toRLPItem(rlpParentNodes);
         RLP.RLPItem[] memory parentNodes = RLP.toList(item);
 
+        //Sha3 of account address - used from traversing from root to desired node
+        bytes memory encodedPath = bytes32ToBytes(keccak256(addr));
```

Comment:
by providing the encoded path in the parameters the `verify` function is general for either an account lookup (which is stored under the sha3 of its address indeed) as well as for other values (ie for storage values inside the accounts storage)